slider.js

```diff
@@ -0,0 +1,109 @@
+const SLIDER_CLASSES = {
+  LIST: ".slider-list",
+  ITEM: ".slider-item",
+  ACTIVE_ITEM: ".slider-item active",
+  PLACEHOLDER: ".placeholder",
+  ADD_SHOW_PLACEHOLDER: "show-placeholder",
+  NAVIGATION_BUTTONS: ".navigation-buttons",
+  ADD_NO_NAVIGATION_BUTTONS: "no-navigation",
+  BTN_PREV: ".btn-prev",
+  BTN_NEXT: ".btn-next",
+};
+
+const createSlider = (function () {
+  let activeSlideIndex;
+
+  const getSlides = (sliderRoot) => {
+    let slides;
+    const list = sliderRoot.querySelector(SLIDER_CLASSES.LIST);
+    slides = list.querySelectorAll(SLIDER_CLASSES.ITEM);
+
+    const showPlaceholder = () => {
+      sliderRoot
+        .querySelector(SLIDER_CLASSES.PLACEHOLDER)
+        .classList.add(SLIDER_CLASSES.ADD_SHOW_PLACEHOLDER);
+    };
+    return slides.length > 0 ? slides : showPlaceholder();
+  };
+
+  const getButtons = (sliderRoot) => {
+    const navigation = sliderRoot.querySelector(
+      SLIDER_CLASSES.NAVIGATION_BUTTONS
+    );
+    const prevButton = navigation.querySelector(SLIDER_CLASSES.BTN_PREV);
+    const nextButton = navigation.querySelector(SLIDER_CLASSES.BTN_NEXT);
+
+    return {
+      prevButton,
+      nextButton,
+    };
+  };
+
+  const getPrevSlideIndex = (activeSlideIndex, slides) => {
+    return activeSlideIndex === 0 ? slides.length - 1 : activeSlideIndex - 1;
+  };
+
+  const getNextSlideIndex = (activeSlideIndex, slides) => {
+    return activeSlideIndex === slides.length - 1 ? 0 : activeSlideIndex + 1;
+  };
+
+  const getPrevSlide = () => {
```

Comment:
by the function name I expect that the function returns prev slide. But it actually changes slide. Also it returns activeSlideIndex (not a slide) and  returning value is not used in place where it's called. There is something wrong. How can it be fixed?